semistruct_test_utils.go

```diff
@@ -0,0 +1,224 @@
+package semistruct
```

Comment:
is the file for testing or the for the package? Because it doesn't end with test the code will be included in the build. Is that what you want? If not change it semistruct_utils_test.go